### PR TITLE
Stop GoAccess errors

### PR DESCRIPTION
**Pull request by Sproglet**
Minor change to delay goaccess for 30 seconds to allow access.log to be created on 1st run and log rotate.

Every logrotate I was still getting a lot of errors from goaccess complaining that the access.log didn't exist. a pause of 10 seconds is enough but 30 to be safe.

Pause is in the goaccess rather than the logrotate as it uses less cpu cycles in this place.

### CODE REVIEW

**Comment by Zoey2936** [comment]
maybe instead check if the file exists?

**Comment by Sproglet** [comment]
> maybe instead check if the file exists?

I did look at that but monitoring/waiting for a file uses a lot more cpu cycles than a sleep so this is better for low powered machines that most home users use.

File: rootfs/usr/local/bin/launch.sh [hunk 1]
```diff
@@ -76,7 +76,7 @@ if [ "$PHP82" = "true" ]; then PHP_INI_SCAN_DIR=/data/php/82/conf.d php-fpm82 -c
 if [ "$PHP83" = "true" ]; then PHP_INI_SCAN_DIR=/data/php/83/conf.d php-fpm83 -c /data/php/83 -y /data/php/83/php-fpm.conf -FOR; fi &
 if [ "$LOGROTATE" = "true" ]; then while true; do touch /data/etc/logrotate.lock; logrotate --verbose --state /data/etc/logrotate.state /etc/logrotate; rm /data/etc/logrotate.lock; sleep 25h; done; fi &
 # shellcheck disable=SC2086
-if [ "$GOA" = "true" ]; then while true; do if [ ! -f /data/etc/logrotate.lock ]; then goaccess --no-global-config --num-tests=0 --tz="$TZ" --date-format="%d/%b/%Y" --time-format="%H:%M:%S" --log-format='[%d:%t %^] %v %h %T "%r" %s %b %b %R %u' --no-ip-validation \
+if [ "$GOA" = "true" ]; then while true; do if [ ! -f /data/etc/logrotate.lock ]; then sleep 30s;  goaccess --no-global-config --num-tests=0 --tz="$TZ" --date-format="%d/%b/%Y" --time-format="%H:%M:%S" --log-format='[%d:%t %^] %v %h %T "%r" %s %b %b %R %u' --no-ip-validation \
```
Comment (Zoey2936) [inline rootfs/usr/local/bin/launch.sh]:
```suggestion
if [ "$GOA" = "true" ]; then while true; do if [ -f /data/nginx/access.log ] && [ ! -f /data/etc/logrotate.lock ]; then goaccess --no-global-config --num-tests=0 --tz="$TZ" --date-format="%d/%b/%Y" --time-format="%H:%M:%S" --log-format='[%d:%t %^] %v %h %T "%r" %s %b %b %R %u' --no-ip-validation \
```
I mean like this

Comment (Zoey2936) [reply rootfs/usr/local/bin/launch.sh]: it will wait 10s between each check of file

Comment (Sproglet) [reply rootfs/usr/local/bin/launch.sh]: its worth a try. I'll update my code and test it. I've been running my code for a week. I'll try yours now.

Comment (Zoey2936) [reply rootfs/usr/local/bin/launch.sh]: does it work for you?

Comment (Sproglet) [reply rootfs/usr/local/bin/launch.sh]: Sorry away from my system at the moment. I did make the change and will check my logs for errors when I get back on Saturday.

Comment (Zoey2936) [reply rootfs/usr/local/bin/launch.sh]: Thanks

Comment (Sproglet) [reply rootfs/usr/local/bin/launch.sh]: Managed to access my server tonight. Logs are rotating with no errors however there is another issue with goaccess that I've found. In the rare case where the server loses power or is shut down before the logrotate is due and doesn't restart till past the time the logrotate is due the logs will be rotated and not ingested into the goaccess DBs.
So one issue fixed and a new one found :)

Comment (Zoey2936) [reply rootfs/usr/local/bin/launch.sh]: Please create a new issue for this, thanks!

**Review by Zoey2936 — APPROVED** [review]

**Pull request merged**